pkg/apis/core/validation/validation.go

```diff
@@ -1770,20 +1770,30 @@ func ValidatePersistentVolume(pv *core.PersistentVolume) field.ErrorList {
 // ValidatePersistentVolumeUpdate tests to see if the update is legal for an end user to make.
 // newPv is updated with fields that cannot be changed.
 func ValidatePersistentVolumeUpdate(newPv, oldPv *core.PersistentVolume) field.ErrorList {
+	newPvClone := newPv.DeepCopy()
+	oldPvClone := oldPv.DeepCopy()
+
 	allErrs := field.ErrorList{}
-	allErrs = ValidatePersistentVolume(newPv)
+	allErrs = ValidatePersistentVolume(newPvClone)
+
+	if utilfeature.DefaultFeatureGate.Enabled(features.ExpandCSIVolumes) {
+		// if oldPV does not have ControllerExpandSecretRef then allow it to be set
+		if oldPvClone.Spec.CSI != nil && oldPvClone.Spec.CSI.ControllerExpandSecretRef == nil {
+			newPvClone.Spec.CSI.ControllerExpandSecretRef = nil
+		}
+	}
 
 	// PersistentVolumeSource should be immutable after creation.
-	if !apiequality.Semantic.DeepEqual(newPv.Spec.PersistentVolumeSource, oldPv.Spec.PersistentVolumeSource) {
+	if !apiequality.Semantic.DeepEqual(newPvClone.Spec.PersistentVolumeSource, oldPvClone.Spec.PersistentVolumeSource) {
 		allErrs = append(allErrs, field.Forbidden(field.NewPath("spec", "persistentvolumesource"), "is immutable after creation"))
 	}
-	newPv.Status = oldPv.Status
```

Comment:
preexisting, but status stomping is done in persistentvolumeStrategy#PrepareForUpdate, we should not have been mutating newPv in ValidatePersistentVolumeUpdate

Reply:
because this is overwritten in PrepareForUpdate, we can drop it from here

Reply:
removed